source/contributing/implementation/datamanipulator.rst

```diff
@@ -0,0 +1,87 @@
+==============================
+Implementing a DataManipulator
+==============================
+
+If you want to properly implement a ``DataManipulator`` you have to:
+
+1. Implement the ``DataManipulator`` itself
+#. Implement the ``ImmutableDataManipulator``
+
+Upon completion of the first steps you'll have to get this done too:
+
+#. Register the ``Key`` in the ``KeyRegistry``
+#. Implement the ``DataProcessor``
+#. Implement the ``ValueProcessor`` for each value being represented by the ``DataManipulator``
+
+1. Implementing the DataManipulator
+===================================
+
+Implementing the ``DataManipulator`` is done by extending AbstractData in the easiest case. Heres a short example:
+
+.. code-block:: java
+
+  public class SpongeVelocityData extends AbstractSingleData<Vector3d, VelocityData, ImmutableVelocityData> implements VelocityData {
+ ...
+  }
+
+Let's assume we have a single value based manipulator, like for example the ``flying`` state on an entity which can be ``true`` or ``false``.
+We abstracted things above, so now we're able to just extend ``AbstractBooleanData``.
+The next step is to provide the actual ``DataManipulator.class``, the ``Keys.YOUR_MANIPULATOR_NAME_HERE`` and the default value.
```

Comment:
Correction should be noted: We must provide the generic `DataManipulator` type that follows the generic of `FlyingData`, which in this case is `FlyingData`. So, in our super constructor, we use `FlyingData.class`. Then the `Key` that identifies the "value" of the flying state is recommended to be `Keys.IS_FLYING`, that way it can always be used.